pkg/provisioningv2/rke2/planner/planner.go

```diff
@@ -873,10 +871,16 @@ func (p *Planner) desiredPlan(controlPlane *rkev1.RKEControlPlane, tokensSecret
 	}
 
 	if isEtcd(entry) {
-		nodePlan, err = p.addEtcdSnapshotListPeriodicInstruction(nodePlan, controlPlane)
+		nodePlan, err = p.addEtcdSnapshotListLocalPeriodicInstruction(nodePlan, controlPlane)
 		if err != nil {
 			return nodePlan, err
 		}
+		if controlPlane != nil && controlPlane.Spec.ETCD != nil && controlPlane.Spec.ETCD.S3 != nil {
+			nodePlan, err = p.addEtcdSnapshotListS3PeriodicInstruction(nodePlan, controlPlane)
+			if err != nil {
+				return nodePlan, err
+			}
+		}
```

Comment:
If s3 snapshots are enabled for a cluster, then we will essentially be doing these allocations twice and error checks twice every time.

I like the idea of doing an `if-else` statement here to avoid that.